runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -344,6 +344,13 @@ public DynamicMessage createDynamicMessageFromJson(Any anyMsg,
         return builder.build();
     }
 
+    public DynamicMessage createDynamicMessage(Any message) throws DescriptorValidationException, InvalidProtocolBufferException {
+        String fullMessageName = getFullMessageName(message);
+        FileDescriptor valueFileDescriptor = getDescriptor(messagesFdProtoNameMap.get(fullMessageName));
+        Descriptors.Descriptor valueDescriptor = valueFileDescriptor.findMessageTypeByName(getMessageName(message));
```

Comment:
cant we reuse 'fullMessageName' from L348 instead of invoking getMessageName() again?

Reply:
No, because they return different results - getMessageName() only returns the className without path while getFullMessageName will return the classname with path.